agent/agent-bootstrap/src/main/java/com/microsoft/applicationinsights/agent/bootstrap/configuration/ConfigurationBuilder.java

```diff
@@ -61,27 +67,70 @@ public static Configuration create(Path agentJarPath) throws IOException {
         Configuration config = loadConfigurationFile(agentJarPath);
         PreviewConfiguration preview = config.instrumentationSettings.preview;
 
-        preview.roleName = overlayWithEnvVar(APPLICATIONINSIGHTS_ROLE_NAME, WEBSITE_SITE_NAME, preview.roleName);
+        preview.roleName = overlayWithEnvVars(APPLICATIONINSIGHTS_ROLE_NAME, WEBSITE_SITE_NAME, preview.roleName);
         preview.roleInstance =
-                overlayWithEnvVar(APPLICATIONINSIGHTS_ROLE_INSTANCE, WEBSITE_INSTANCE_ID, preview.roleInstance);
+                overlayWithEnvVars(APPLICATIONINSIGHTS_ROLE_INSTANCE, WEBSITE_INSTANCE_ID, preview.roleInstance);
 
-        if (!jmxMetricExisted(preview.jmxMetrics, "java.lang:type=Threading", "ThreadCount")) {
-            JmxMetric threadCountJmxMetric = new JmxMetric();
-            threadCountJmxMetric.objectName = "java.lang:type=Threading";
-            threadCountJmxMetric.attribute = "ThreadCount";
-            threadCountJmxMetric.display = "Current Thread Count";
-            preview.jmxMetrics.add(threadCountJmxMetric);
+        loadSamplingRateEnvVar(preview);
+        loadLogCaptureEnvVar(preview);
+        loadJmxMetrics(preview);
+
+        return config;
+    }
+
+    private static void loadSamplingRateEnvVar(PreviewConfiguration preview) {
+        final String samplingRate = overlayWithEnvVar(APPLICATIONINSIGHTS_SAMPLING_RATE, "100");
+        if (samplingRate != "100") {
+            configurationMessages.add(new ConfigurationMessage("Update default sampling rate from 100 to {}", samplingRate));
+            FixedRateSampling fixedRateSampling = new FixedRateSampling();
+            fixedRateSampling.percentage = new Double(samplingRate);
+            Sampling sampling = new Sampling();
+            sampling.fixedRate = fixedRateSampling;
+            preview.sampling = sampling;
         }
+    }
 
-        if (!jmxMetricExisted(preview.jmxMetrics, "java.lang:type=ClassLoading", "LoadedClassCount")) {
-            JmxMetric classCountJmxMetric = new JmxMetric();
-            classCountJmxMetric.objectName = "java.lang:type=ClassLoading";
-            classCountJmxMetric.attribute = "LoadedClassCount";
-            classCountJmxMetric.display = "Loaded Class Count";
-            preview.jmxMetrics.add(classCountJmxMetric);
+    private static void loadLogCaptureEnvVar(PreviewConfiguration preview) {
+        final String logCaptureThreshold = overlayWithEnvVar(APPLICATIONINSIGHTS_LOG_CAPTURE_THRESHOLD, "INFO");
+        if (logCaptureThreshold != "INFO") {
+            configurationMessages.add(new ConfigurationMessage("Update default log capture threshold from INFO to {}", logCaptureThreshold));
+            final Map<String, Object> threshold = new HashMap<String, Object>() {{
+                put("threshold", logCaptureThreshold);
+            }};
+            preview.instrumentation = new HashMap<String, Map<String, Object>>() {{
+                put("logging", threshold);
+            }};
```

Comment:
currently this will overwrite existing instrumentation settings map provided from json file

if u change:
`public Map<String, Map<String, Object>> instrumentation = Collections.emptyMap()`
to
`public Map<String, Map<String, Object>> instrumentation = new HashMap<>()`

then can simplify:

```suggestion
            Map<String, Object> logging = preview.instrumentation.get("logging");
            if (logging == null) {
                logging = new HashMap<>();
            }
            logging.put("threshold", logCaptureThreshold);
```